File: E-commerce web application/Backend/Application/UserCase/Product/ProductService.cs
using Application.Interface;
using Application.Response;

namespace Application.UserCase.Product
{
    public class ProductService : IProductService
    {
        private readonly IProductQuery _query;

        public ProductService(IProductQuery query)
        {
            _query = query;
        }

        public async Task<IEnumerable<ProductResponse>> GetAll(string name, bool sort)
        {
            var result = await _query.GetAll(name, sort);
            if (result.Count==0)
            {
                throw new NonExistentNameException();
            }
            return result.Select(p=> new ProductResponse
            {
                ProductId = p.ProductId,
                Name =p.Name,
                Brand=p.Brand,
                Code=p.Code,
                Price=p.Price,
                Image=p.Image,
                Description=p.Description
            });
        }
        public Task<ProductResponse> GetProduct(int id)
        {
            var product = _query.GetProduct(id);
            if (product==null)
            {
                throw new NonExistentIDException();
            }
            return Task.FromResult(new ProductResponse
            {
                ProductId=product.ProductId,
                Name=product.Name,
                Brand=product.Brand,
                Price=product.Price,
                Code=product.Code,
                Image=product.Image,
                Description=product.Description
            });
        }
    }
}

File: E-commerce web application/Frontend/js/services/ProductService.js
import { ApiClient } from "../api/ApiClient.js";

export class ProductService {
  static async getAll({ name = "", sort = false } = {}) {
    const params = new URLSearchParams();
    if (name) params.set("name", name);
    params.set("sort", sort);
    return ApiClient.get(`/product?${params.toString()}`);
  }

  static async getById(id) {
    return ApiClient.get(`/product/${id}`);
  }
}